examples/tv-casting-app/android/App/app/src/main/jni/cpp/core/MatterCastingPlayer-JNI.cpp

```diff
@@ -115,16 +128,76 @@ JNI_METHOD(jobject, verifyOrEstablishConnection)
         }
     };
 
-    // TODO: In the following PRs. Add optional CommissionerDeclarationHandler callback parameter for the Commissioner-Generated
-    // passcode commissioning flow.
+    // jCommissionerDeclarationCallback is optional
+    if (jCommissionerDeclarationCallback == nullptr)
+    {
+        ChipLogProgress(AppServer,
+                        "MatterCastingPlayer-JNI::verifyOrEstablishConnection() optional jCommissionerDeclarationCallback was not "
+                        "provided by the client");
+    }
+    else
+    {
+        MatterCastingPlayerJNIMgr().mCommissionerDeclarationHandler.SetUp(env, jCommissionerDeclarationCallback);
+    }
+
+    auto commissionerDeclarationCallback = [](const chip::Transport::PeerAddress & source,
+                                              chip::Protocols::UserDirectedCommissioning::CommissionerDeclaration cd) {
+        ChipLogProgress(AppServer, "MatterCastingPlayer-JNI::verifyOrEstablishConnection() CommissionerDeclarationCallback()");
+        cd.DebugLog();
+
+        char addressStr[chip::Transport::PeerAddress::kMaxToStringSize];
+        source.ToString(addressStr, sizeof(addressStr));
+        ChipLogProgress(AppServer,
+                        "MatterCastingPlayer-JNI::verifyOrEstablishConnection() CommissionerDeclarationCallback() source: %s",
+                        addressStr);
+
+        // Call the Java CommissionerDeclarationCallback if it was provided by the client.
+        if (!MatterCastingPlayerJNIMgr().mCommissionerDeclarationHandler.IsSetUp())
+        {
+            ChipLogError(AppServer,
+                         "MatterCastingPlayer-JNI::verifyOrEstablishConnection() CommissionerDeclarationCallback() received from "
+                         "but Java callback is not set");
+        }
+        else
+        {
+            MatterCastingPlayerJNIMgr().mCommissionerDeclarationHandler.Handle(cd);
+        }
+    };
```

Comment:
This JNI\_METHOD has gotten too long. Can you move this new logic to a separate private function and call it from here?

Reply:
ok, will bring out ConnectCallback and CommissionerDeclarationCallback.